app/src/main/java/de/xikolo/utils/LanguageUtil.kt

```diff
@@ -1,18 +1,40 @@
 package de.xikolo.utils
 
+import android.content.Context
+import android.os.Build
 import java.util.Locale
 
 object LanguageUtil {
+    var deviceLanguage: String = Locale.getDefault().language.takeIf { it != "zh" } ?: "cn"
+
     @OptIn(ExperimentalStdlibApi::class)
     fun toNativeName(tag: String): String {
-        val languageTag =
-            if (tag == "cn") { // work-around for bug in the backend
-                "zh"
-            } else {
-                tag
-            }
+        val languageTag = correctLanguageTag(tag)
 
         val locale = Locale.forLanguageTag(languageTag)
         return locale.getDisplayLanguage(locale).capitalize(locale)
     }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    fun toLocaleName(context: Context, tag: String): String {
+        val languageTag = correctLanguageTag(tag)
+        val locale = Locale.forLanguageTag(languageTag)
+
+        val deviceLocale =
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                context.resources.configuration.locales.get(0)
```

Comment:
You are using the `context` here to fetch the device language, yet in `deviceLanguage` you are using `Locale.getDefault()`. I think this should be consistent.

Reply:
Found a solution to use even less code :) maybe you can have a look